File: python/questions/minSubArrayLen.py
from typing import List

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        ws = 0
        l = 0
        ans = len(nums) + 1
        for r, v in enumerate(nums):
            ws += v
            if ws >= target:
                while ws >= target:
                    ws -= nums[l]
                    l += 1
                ans = min(ans, r-l+2)
        return ans if ans!=len(nums)+1 else 0

File: cpp/questions/minSubArrayLen.cpp
#include <vector>
#include <climits>

using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int ns = nums.size();
        int l = 0;
        int r = 0;
        int ws = 0;
        int ans = INT_MAX;
        for (; l<ns; ++l) {
            for (; r<ns && ws<target; ++r) {
                ws += nums[r];
            }
            if (ws >= target) {
                ans = std::min(ans, r-l);
            } else if (r == ns && ws<target) {
                return ans==INT_MAX ? 0 : ans;
            }
            ws -= nums[l];
        }
        return ans==INT_MAX ? 0 : ans;
    }
};
